djstripe/contrib/rest_framework/serializers.py

```diff
@@ -8,12 +8,197 @@
 """
 
 from rest_framework import serializers
-from rest_framework.serializers import ModelSerializer
+from rest_framework.exceptions import MethodNotAllowed
+from rest_framework.serializers import ModelSerializer, ValidationError
 
-from djstripe.models import Subscription
+from djstripe.enums import SubscriptionStatus
+from djstripe.models import Plan, Subscription
+from djstripe.settings import CANCELLATION_AT_PERIOD_END
 
+from .mixins import AutoCustomerModelSerializerMixin
 
-class SubscriptionSerializer(ModelSerializer):
+
+class SubscriptionSerializer(AutoCustomerModelSerializerMixin, ModelSerializer):
+    """A base serializer used for the Subscription model."""
+
+    class Meta:
+        model = Subscription
+        exclude = ["default_tax_rates"]
+
+    # StripeModel fields
+    id = serializers.CharField(required=False)
+    created = serializers.DateTimeField(required=False)
+    metadata = serializers.JSONField(required=False)
+    description = serializers.CharField(required=False)
+
+    # Subscription-specific fields
+    application_fee_percent = serializers.DecimalField(
+        required=False, max_digits=5, decimal_places=2
+    )
+    billing_cycle_anchor = serializers.DateTimeField(required=False)
+    cancel_at_period_end = serializers.NullBooleanField(required=False)
+    canceled_at = serializers.DateTimeField(required=False)
+    collection_method = serializers.CharField(required=False)
+    current_period_end = serializers.DateTimeField(required=False)
+    current_period_start = serializers.DateTimeField(required=False)
+    customer = serializers.SlugField(required=False, source="customer.id")
+    days_until_due = serializers.IntegerField(required=False)
+    default_payment_method = serializers.SlugField(
+        required=False, source="default_payment_method.id"
+    )
+    default_source = serializers.SlugField(required=False, source="default_source.id")
+    discount = serializers.JSONField(required=False)
+    ended_at = serializers.DateTimeField(required=False)
+    next_pending_invoice_item_invoice = serializers.DateTimeField(required=False)
+    pending_invoice_item_interval = serializers.JSONField(required=False)
+    pending_setup_intent = serializers.SlugField(
+        required=False, source="pending_setup_intent.id"
+    )
+    pending_update = serializers.JSONField(required=False)
+    plan = serializers.SlugField(required=False, source="plan.id")
+    quantity = serializers.IntegerField(required=False)
+    start = serializers.DateTimeField(required=False)
+    start_date = serializers.DateTimeField(required=False)
+    status = serializers.CharField(required=False)
```

Comment:
These fields should be defined automatically from the model via `ModelSerializer`.
Otherwise the fields should provide all of the same metadata as the model, which would be prone to breaking when the model changes in the future.

When I apply this patch, by OpenAPI schema components all degrade, field descriptions are lost, enums are gone, validation is wrong, etc.